src/Stryker.Core/Stryker.Core.UnitTest/MutantFilters/IgnoredMethodMutantFilterTests.cs

```diff
@@ -169,15 +176,82 @@ private void TestMethod()
         }
 
         [Theory]
+        [InlineData("Bar.Foo.Dispose", true)]
+        [InlineData("Bar.*.Dispose", true)]
+        [InlineData("Foo.Dispose*", true)]
+        [InlineData("Foo.Dispos*", true)]
+        [InlineData("*Foo.Dispose", true)]
+        [InlineData("F*.Dispose", true)]
+        [InlineData("*o.Dispose", true)]
+        [InlineData("*o.D*se", true)]
+        [InlineData("*.*", true)]
+        [InlineData("Foo.*", true)]
+        [InlineData("Foo*Dispose", false)]
+        [InlineData("Bar.Foo", false)]
+        [InlineData("Bar*", false)]
+        [InlineData("Bar.", false)]
+        [InlineData("Bar.*", false)]
+        [InlineData("Foo", false)]
+        [InlineData("Foo.", false)]
+        [InlineData("*.*.*.*", false)]
+        public void ShouldFilterInvocationWithQualifiedMemberName(string ignoredMethodName, bool shouldSkipMutant)
+        {
+            // Arrange
+            var source = @"
+public class IgnoredMethodMutantFilter_NestedMethodCalls
+{
+    private void TestMethod()
+    {
+        Bar // comment
+            .Foo.Dispose();
+    }
+}";
+            var baseSyntaxTree = CSharpSyntaxTree.ParseText(source).GetRoot();
+            var originalNode = baseSyntaxTree.FindNode(new TextSpan(source.IndexOf("Dispose"), 1));
+
+            var mutant = new Mutant
+            {
+                Mutation = new Mutation
+                {
+                    OriginalNode = originalNode,
+                }
+            };
+
+            var options = new StrykerOptions(ignoredMethods: new[] { ignoredMethodName });
+
+            var sut = new IgnoredMethodMutantFilter();
+
+            // Act
+            var filteredMutants = sut.FilterMutants(new[] { mutant }, null, options);
+
+            // Assert
+            if (shouldSkipMutant)
+            {
+                filteredMutants.ShouldNotContain(mutant);
+            }
+            else
+            {
+                filteredMutants.ShouldContain(mutant);
+            }
+        }
+
```

Comment:
I'm missing a test case here that I think won't work yet.

We now prove that we can ignore methods using this option. But I doubt we ignore ONLY the correct method. With the current implementation I think we ignore all mutations in a file when one of the methods matches the regex. 

I'd like to see some tests with multiple mutants where only 1 is ignored using the option. All others should not be ignored

Reply:
I added a test with multiple mutants in a single file, but it passed without any code changes. Let me know if it assuages your concerns or if there's some other tests we should add.

Reply:
Hmm well than it seems that my concern was unfairly :) 

This test takes away all of my concerns so thats good.